jade/src/com/jetbrains/plugins/jade/js/JSInJadeEmbeddedContentImpl.java

```diff
@@ -4,13 +4,13 @@
 import com.intellij.extapi.psi.ASTWrapperPsiElement;
 import com.intellij.lang.ASTNode;
 import com.intellij.lang.Language;
-import com.intellij.lang.javascript.psi.JSEmbeddedContent;
+import com.intellij.lang.javascript.psi.*;
 import com.intellij.lang.javascript.psi.controlflow.JSControlFlowService;
 import com.intellij.psi.tree.IElementType;
 import com.jetbrains.plugins.jade.psi.JadeTokenTypes;
 import org.jetbrains.annotations.NotNull;
 
-public class JSInJadeEmbeddedContentImpl extends ASTWrapperPsiElement implements JSEmbeddedContent {
+public class JSInJadeEmbeddedContentImpl extends ASTWrapperPsiElement implements JSElement, JSControlFlowScope {
```

Comment:
This is needed to let the declarations "escape" from the current element into the file, but breaks completion for all variables defined in JS_IN_JADE. I think this behaviour comes from somewhere in `JSSymbolCompletionUtil.processCompeteMatchedElements`, but I can't step into there.

`JadeCompletionTest.testVariable1` fails because of this

Reply:
Added a completion contributor, which is _a_ solution, but probably not _the_ solution